src/components/EmailNotifications/EmailNotificationsForm.js

```diff
@@ -0,0 +1,302 @@
+                  type="email"
+                  wide
+                  onChange={handleEmailAddressChange}
+                  placeholder="email@address.com"
+                  onBlur={handleEmailAddressBlur}
+                />
+              </Field>
+
+              {!isModal && (
+                <div
+                  css={`
+                    display: flex;
+                    align-items: flex-end;
+                    margin-left: ${2 * GU}px;
+                  `}
+                >
+                  <Button
+                    mode="strong"
+                    disabled={emailInvalid || !termsAccepted}
+                    onClick={handleOnSubscribeToNotifications}
+                    size="medium"
+                  >
+                    {existingEmail ? 'Update' : 'Subscribe'}
+                  </Button>
+                </div>
+              )}
+            </div>
+
+            {emailInvalid && (
+              <div>
+                <p
+                  css={`
+                    color: ${theme.negative};
+                    ${textStyle('body4')};
+                    text-align: left;
+                    height: 0;
+                    margin-top: ${0.5 * GU}px;
+                  `}
+                >
+                  Please enter a valid email address.
+                </p>
+              </div>
+            )}
+          </div>
+        </div>
+        <LegalTermsAndPolicy
+          termsAccepted={termsAccepted}
+          setTermsAccepted={setTermsAccepted}
+        />
+
+        {isModal && (
+          <div
+            css={`
+              display: flex;
+              justify-content: space-between;
+              flex-direction: ${compactMode ? 'column' : 'row'};
+              width: 100%;
+              margin-bottom: ${1.5 * GU}px;
+              margin-top: ${3 * GU}px;
+            `}
+          >
+            <ActionButtons compactMode={compactMode} onClick={handleOnOptOut}>
+              Opt out
+            </ActionButtons>
+            <ActionButtons
+              compactMode={compactMode}
+              mode="strong"
+              disabled={emailInvalid || !termsAccepted}
+              onClick={handleOnSubscribeToNotifications}
+            >
+              Subscribe
+            </ActionButtons>
+          </div>
+        )}
+      </div>
+    </>
+  )
+}
+
+function TextContent({ update, account }) {
+  const theme = useTheme()
+  const content = update
+    ? `Enter a new email address for your account ${account}. We will continue sending email notifications to the current email address until you verify this new email address.`
+    : `Associate an email address to your account ${account}, so you can get notifications from all Aragon Court events.`
+
+  return (
+    <span
+      css={`
+        ${textStyle('body2')};
+        color: ${theme.surfaceContentSecondary};
+        margin-top: ${1.5 * GU}px;
+      `}
+    >
+      {transformAddresses(content, (part, isAddress, index) =>
```

Comment:
Mentioned elsewhere, this pattern was really meant for a component receiving generic text. If this is a pattern that looks really useful, we could make a component out of this (e.g. `AddressifyText` or etc), but at the same time, we could also just inline the copy and `IdentityBadge` like:

```js
return (
  <span>
    {updated ? (
      <>
        <span>text</span>
        <IdentityBadge />
        <span>more text</span>
      </>
    ) : (
      <>
        <span>other text</span>
        <IdentityBadge />
        <span>more text</span>
      </>
    )}
  </span>
)
```

Reply:
any good reason to change it since we have this known pattern, known function accross all our projects? i would be more like create a component than for changing it

Reply:
not sure why we should avoid using an existing function that we already have?

Reply:
This function wasn't intended to be used like this, and while sometimes you will find interesting new ways to re-use a prior component that really proves your initial API's elegance, other times it will feel like an abuse. This is the latter.

For me, using this function here obfuscates the intention. If we're going to to interpolate a known string to place the address and then do another transform on it, why don't we just render the string and IdentityBadge directly?

The other option is to wrap the `transformAddresses()` into a generic component that receives generic input and renders a combination of strings and IdentityBadges (e.g. [this component](https://github.com/aragon/aragon-apps/blob/master/apps/voting/app/src/components/VoteText.js#L10)). This places the responsibility of handling this transformation onto a sole component (an "abstraction point") that I can then separately test and reason about.